### ISSUE TITLE
Using Maven option `-f` with a relative path breaks retrieving copyright year from git

### ISSUE THREAD

**Issue by jgerken** [mycila/license-maven-plugin#138]
We are using the property `license.git.copyrightYears` from `license-maven-plugin-git` in our license files. This normally works well, e.g. creating a copyright span `2011-2018`. However, when the Maven build is triggered from a different directory using `mvn -f <path to directory containing the POM> <maven phase>` with a relative path (e.g. `../core`) only the inception year is returned as copyright span (e.g. `2011`). When an absolute path is used, the copyright span is correct again (e.g. `2011-2018`).

**Comment by mathieucarbou** [comment]
Hi @jgerken, 
Thanks for the report. I would need some help for this one because I do not have enough time to troubleshoot it.

**Comment by jgerken** [comment]
I did further tests and it seems it only happens in multi module projects when you build one module while in the directory of another module, e.g.

```
mvn -f ../B process-sources
```
called from the `project/modules/A` directory in the following multi module project:

```
project
|-- modules
|   |-- A
|   |   `-- pom.xml
|   `-- B
|       `-- pom.xml
`-- pom.xml
```

In such cases, the repository directories and all documents contain the part `project/modules/A/../B` which breaks the `GitPathResolver`. In the end, everything trickles down to the `basedir` passed to the `AbstractLicenseMojo` already being not in its canonical form.

The problem can be corrected at multiple locations in the source code. Performance wise, the best position would be directly in `AbstractLicenseMojo`. Since I have never written a Maven plugin myself, I have no idea if the handler of the `@Parameter` annotation would call a `setBasedir` method and which signature it must have. If this is not possible, it could be corrected directly at the beginning of the `execute` method, e.g. like this:

```java
try {
    this.basedir = this.basedir.getCanonicalFile();
} catch (final IOException e) {
    throw new MojoFailureException("Could not get canonical path of project.basedir", e);
}
```

Additionally, using a setter would require changes to a lot of tests because they rely on public access to the `basedir` member.

I could create a PR using the try-catch version if it is ok for you. I just need to write some tests for the git plugin part to have it automatically tested (I did only a manual prove of concept test).

**Comment by mathieucarbou** [comment]
@jgerken is the issue still there ? is your proposal to open a PR still valid ?
Thanks!

**Comment by jgerken** [comment]
@mathieucarbou yes, the issue is still there with 4.0.rc2. My proposed fix for 3.0 does not compile with current master - the `baseDir` variable was renamed to `defaultBaseDir` due to the new license sets I guess. I think I need to move my fix to the sets and it should still work. Need to have a deeper look into the issue. And yes, my proposal to open a PR is still valid. I am not sure if I find the time this week but I should be able to open a PR in this month.

**Comment by jgerken** [comment]
The Windows test gave me a little trouble until I noticed that paths are always logged with `/` as separator. Now the PR is ready for review :)
